src/resource/solidDataset.ts

```diff
@@ -81,8 +81,71 @@ export function createSolidDataset(): SolidDataset {
 }
 
 /**
- * @hidden This interface is not exposed yet; we'll first have to put it through
- *         its paces by trying it out with other parsers than n3 as well.
+ * A Parser takes a string and generates {@link https://rdf.js.org/data-model-spec/|RDF/JS Quads}.
+ *
+ * By default, `@inrupt/solid-client` only includes a parser for `text/turtle`,
+ * which is the default RDF serialisation provided by Solid-compliant servers.
+ * However, if you intend to use it to retrieve RDF data from other sources,
+ * you might need support for additional serialisations. You can do so by
+ * supplying your own parsers for those serialisations.
+ *
+ * A parser needs to provide two things:
+ * - a way to return the parsed output to `@inrupt/solid-client`. This can be
+ *   done via `onQuad`, `onError` and `onComplete`.
+ * - a way to accept the serialised input. This is provided via `parse`.
+ *
+ * Before attempting to parse a Resource, `@inrupt/solid-client` will register
+ * its callback handlers with your parser by passing them to the following
+ * methods:
+ * - `onQuad`: this callback should be called once for every Quad your parser
+ *   encounters in the serialised input.
+ * - `onError`: this callback should be called whenever your parser finds it is
+ *   unable to parse the given input.
+ * - `onComplete`: when your parser will no longer produce new Quads (i.e. it
+ *   has reached the end of the given input), completion should be indicated by
+ *   calling this callback.
+ *
+ * Finally, your parser should provide a `parse` method. After the above
+ * callbacks are registered, this method will be called with the serialised
+ * input and an object containing metadata about the input Resource, which can
+ * be read using functions like [[getSourceUrl]] and [[getContentType]].
+ *
+ * As an example, a parser that reads an RDFa serialisation using the
+ * [rdfa-streaming-parser](https://www.npmjs.com/package/rdfa-streaming-parser)
+ * library could look like this:
```

Comment:
? Maybe shorten it and we can put a bit more color in the reference docs instead?  But absolutely unwedded to my own suggestion - so feel free to ignore.

```suggestion
 * 
 * The type Parser is a type for specifying your own parsers to
 * handle RDF serialisation other than `text/turtle`, which is 
 * handled by `@inrupt/solid-client`'s default parser.
 *
 * Parser has the following properties:
 * 
 * - `onQuad`: Registers the `@inrupt/solid-client` callback for handling each Quad parsed from the serialised input.
 * - `onError`: Registers the `@inrupt/solid-client` callback for handling errors.
 * - `onComplete`: Registers the `@inrupt/solid-client` callback for handling parsing completion.
 * - `parse`: Accepts the serialised input string and an object containing the input Resource's metadata.
 *    The input metadata can be retrieved by using functions like [[getSourceUrl]] and [[getContentType]].
 * 
 * For example, the following defines a parser that reads an RDFa serialisation 
 * using the [rdfa-streaming-parser](https://www.npmjs.com/package/rdfa-streaming-parser) library:
```

Reply:
I tried to tighten it up a bit based on your suggestion, with slight rewordings to be technically correct and/or hopefully support the mental model of callbacks a bit more: ab33e0dd

Reply:
nice! works for me 😄